File: 1768-merge-strings-alternately/1768-merge-strings-alternately.c


char * mergeAlternately(char * word1, char * word2){
    int i = 0, j = 0, count = 0, len = 0;
    char* mergedString = (char*)malloc(sizeof(char) * (strlen(word1) + strlen(word2) + 1));


    while ((i != strlen(word1)) && (j != strlen(word2))) {
        if (count == 0) {
            mergedString[len++] = word1[i++];
            count = 1;
        } else {
            mergedString[len++] = word2[j++];
            count = 0;
        }
    }

    while (i != strlen(word1)) {
        mergedString[len++] = word1[i++];
    }

    while (j != strlen(word2)) {
        mergedString[len++] = word2[j++];
    }

    // Add the null terminator at the end
    mergedString[len] = '\0';

    return mergedString;
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution(object):
    def mergeAlternately(self, word1, word2):
        i=0
        j=0
        new_string=""
        len_word1=len(word1)
        len_word2=len(word2)
        while(i<len_word1 and j<len_word2):
            new_string+=word1[i]+word2[j]
            i+=1
            j+=1
        new_string+=word1[i:]
        new_string+=word2[j:]
        return new_string